### PR TITLE
Function to extract list of mz ranges into IonSeries

### PR DESCRIPTION
**Pull request by robinschmid**
Here I suggest adding an interface TaskFunction. Multiple times, I struggled to reuse code because I wanted to listen to a parent task and update the progress....

This only contains:
- T calculate();
- double getFinishedPercentage()
- get/setParentTask
- isCanceled

### CODE REVIEW

**Comment by robinschmid** [comment]
@ansgarkorf can you try the pseudoscan visualizer?
I updated the class

File: src/main/java/io/github/mzmine/modules/dataprocessing/featdet_extract_mz_ranges/ExtractMzRangesIonSeriesFunction.java [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+      currentScan++;
+      processedScans++;
+
+      // Canceled?
+      if (isCanceled()) {
+        return new BuildingIonSeries[0];
+      }
+      // check value for tree and for all next trees in range
+      int nDataPoints = dataAccess.getNumberOfDataPoints();
+      for (int dp = 0; dp < nDataPoints; dp++) {
```
Comment (SteffenHeu) [inline src/main/java/io/github/mzmine/modules/dataprocessing/featdet_extract_mz_ranges/ExtractMzRangesIonSeriesFunction.java:125]: switch the loops and binary search jump to the mz of the first range. could even jump the index again when going from range to range.

**Review by SteffenHeu — COMMENTED** [review]
looks good. only small comments. 
Also, some of the new interfaces don't have javadoc. please add a description

File: src/main/java/io/github/mzmine/datamodel/featuredata/impl/BuildingIonSeries.java [hunk 2]
```diff
@@ -178,6 +179,31 @@ public boolean addValue(int scanIndex, double mz, double intensity) {
     };
   }
 
+  @Override
+  public double[] getIntensityValues(final double[] dst) {
+    return intensities;
```
[the remark this answers is not in the document]
Comment (robinschmid) [reply src/main/java/io/github/mzmine/datamodel/featuredata/impl/BuildingIonSeries.java:184]: This method is so annoying :D
I need to look into this.

[the remark this answers is not in the document]
Comment (robinschmid) [reply src/main/java/io/github/mzmine/datamodel/featuredata/impl/BuildingIonSeries.java:184]: Unsupported exception now. Should use the direct accessors

**Pull request merged**